activerecord/lib/active_record/collection_cache_key.rb

```diff
@@ -0,0 +1,27 @@
+module ActiveRecord
+  module CollectionCacheKey
+
+    # Generates a cache key for the records in the given collection.
+    # See <tt>ActiveRecord::Relation#cache_key</tt> for details.
+    def collection_cache_key(collection = all, timestamp_column = :updated_at)
+      query_signature = Digest::MD5.hexdigest(collection.to_sql)
+      key = "#{collection.model_name.cache_key}/query-#{query_signature}"
+
+      size, timestamp = if collection.loaded?
+        [collection.size, collection.collect(&timestamp_column).compact.max]
+      else
+        column_type = collection.klass.type_for_attribute(timestamp_column.to_s)
```

Comment:
Not in cases in which you call `ActiveRecord::Base.collection_cache_key(collection)` [like this one](https://github.com/afcapel/rails/blob/c6f14e573f0cbd90905f9cc4c781d0b20f95b463/activerecord/test/cases/collection_cache_key_test.rb#L19). It's a bit annoying, but since a class method is public I think we need to support that kind of calls.